system/jlib/jcontainerized.cpp

```diff
@@ -63,13 +62,45 @@ void deleteResource(const char *componentName, const char *resourceType, const c
     remove(k8sResourcesFilename);
 }
 
+bool checkExitCodes(StringBuffer &output, const char *podStatuses)
+{
+    const char *startOfPodStatus = podStatuses;
+    while (*startOfPodStatus)
+    {
+        const char *endOfPodStatus = strchr(startOfPodStatus, '|');
+        StringBuffer podStatus;
+        if (endOfPodStatus)
+            podStatus.append((size_t)(endOfPodStatus-startOfPodStatus), startOfPodStatus);
+        else
+            podStatus.append(startOfPodStatus);
+        StringArray fields;
+        fields.appendList(podStatus, ",");
+        if (3 == fields.length()) // should be 3 fields {<exitCode>,<"initContainer"|"container">,<name>}
+        {
+            const char *exitCodeStr = fields.item(0);
+            if (strlen(exitCodeStr))
+            {
+                unsigned exitCode = atoi(exitCodeStr);
+                if (exitCode) // non-zero = failure
+                {
+                    output.appendf(" %s '%s' failed with exitCode = %u", fields.item(1), fields.item(2), exitCode);
+                    return true;
```

Comment:
can more than one fail?  Is it worth continuing?  (I suspect the answer is no)

Reply:
no, 1st will cause the pod to fail.